src/nets/e2e_asr_attctc.py

```diff
@@ -589,6 +608,13 @@ def recognize_beam(self, h, recog_args, char_list):
         hyp = {'score': 0.0, 'yseq': [y], 'c_prev': c_list, 'z_prev': z_list, 'a_prev': a}
         hyps = [hyp]
         ended_hyps = []
+        if lpz is not None:
+            ctc_prefix_score = CTCPrefixScore(lpz, 0, self.eos, self.xp)
+            hyp['ctc_prev'] = ctc_prefix_score.initial_state()
+            score_type = 'joint_score'
```

Comment:
It looks good, but what is your motivation of keeping both score and joint_score? If this is necessary, it is fine. If we keep these scores separately, another option is to keep `att_score` (current `score`) and `ctc_score`. Then, we may also additionally keep `total_score` so that we can extract all of these three scores, which will be good for a readability, and also good for debugging (we can provide both att, ctc, and total scores). Then, we could also add another scores with a same manner (e.g.,  `rnnlm_score`)

Reply:
We need to keep both scores since CTC scores cannot be computed as "alpha(h) = alpha(g) + log P(c|g)" I agree with the idea keeping individual scores if overhead is not a problem.
Another possible way is that we keep only ctc_score in addition to the score, and add "ctc_weight * (ctc_score - previous_ctc_score)" to the score.

Reply:
OK. Then, if we'll not have some overheads, I prefer to keep both scores, but it's up to you.